tools/testdebug/cmd/load_etcd.go

```diff
@@ -0,0 +1,172 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	coreosetcdclient "github.com/coreos/etcd/client"
+	"github.com/spf13/cobra"
+
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/cmd/admin/policy"
+	"github.com/openshift/origin/pkg/cmd/flagtypes"
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	"github.com/openshift/origin/pkg/cmd/server/etcd"
+	"github.com/openshift/origin/pkg/cmd/server/etcd/etcdserver"
+	"github.com/openshift/origin/pkg/cmd/server/origin"
+	"github.com/openshift/origin/pkg/cmd/server/start"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+)
+
+const RecommendedLoadEtcdDumpName = "start-api"
+
+type DebugAPIServerOptions struct {
+	Out io.Writer
+
+	EtcdDumpFile string
+	AllowAll     bool
+}
+
+func NewDebugAPIServerCommand() *cobra.Command {
+	o := &DebugAPIServerOptions{Out: os.Stdout}
+
+	cmd := &cobra.Command{
+		Use:   RecommendedLoadEtcdDumpName + " etcd_dump.json",
+		Short: "Start API server using etcddump",
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(args))
+
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().BoolVar(&o.AllowAll, "allow-all", true, "change policy to grant system:authenticated cluster-admin powers")
+
+	flagtypes.GLog(cmd.PersistentFlags())
+
+	return cmd
+}
+
+func (o *DebugAPIServerOptions) Complete(args []string) error {
+	if len(args) != 1 {
+		return errors.New("etcd_dump.json file is required")
+	}
+
+	o.EtcdDumpFile = args[0]
+
+	return nil
+}
+
+func (o *DebugAPIServerOptions) Run() error {
+	masterConfig, err := testserver.DefaultMasterOptionsWithTweaks(true, true)
```

Comment:
> naming those variables would look better than true, true, especially as the function name is WithTweaks and which tweaks are never specified...

It's a bit of a hack.  Its "options for the options generator of the options".  Ugly, but it's in test code, so refactoring it isn't high on my list.